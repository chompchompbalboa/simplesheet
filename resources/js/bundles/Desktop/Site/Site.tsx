//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React, { useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'

import { IAppState } from '@/state'

import SiteLoginForm from '@desktop/Site/SiteLoginForm'
import SiteRegisterForm from '@desktop/Site/SiteRegisterForm'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const Site = () => {
  
  // Redux
  const userColorPrimary = useSelector((state: IAppState) => state.user.color.primary)
  
  // State
  const [ isLoginFormDocked, setIsLoginFormDocked ] = useState(false)
  const [ isRegisterFormDocked, setIsRegisterFormDocked ] = useState(true)
  
  // Effects
  useEffect(() => {
    addEventListener('scroll', handleScroll)
    return () => removeEventListener('scroll', handleScroll)
  }, [])
  
  // Handle Scroll
  const handleScroll = () => {
    // If the site container is no longer visible, dock it to the top
    if(window.scrollY >= window.innerHeight) {
      setIsLoginFormDocked(true)
      removeEventListener('scroll', handleScroll)
    }
  }

  // Handle Call To Action Click
  const handleCallToActionClick = () => {
    setIsRegisterFormDocked(false)
  }
  

  return (
    <Container
      isLoginFormDocked={isLoginFormDocked}>
      <SiteSplash
        userColorPrimary={userColorPrimary}>
        <Header>
          <HeaderName>
            tasksheet
          </HeaderName>
          <HeaderLinks>
            <HeaderLink>30-day free trial<br/>$5 per month or $100 for lifetime access</HeaderLink>
          </HeaderLinks>
        </Header>
        <Splash>
          <Name>tasksheet</Name>
          <Motto>The spreadsheet that makes it easy to organize your to-dos</Motto>
          <Divider />
          <LoginRegisterContainer>
            <SiteLoginForm />
          </LoginRegisterContainer>
        </Splash>
        <SpreadsheetIcon
          src={environment.assetUrl + 'images/spreadsheet.png'}/>
      </SiteSplash>
      <CallToAction
        isLoginFormDocked={isLoginFormDocked}
        onClick={handleCallToActionClick}>
        Click here to sign up for a free 30-day trial
      </CallToAction>
      <RegisterFormContainer
        isRegisterFormDocked={isRegisterFormDocked}
        userColorPrimary={userColorPrimary}>
        <SiteRegisterForm />
      </RegisterFormContainer>
      <AppOverlay></AppOverlay>
    </Container>
  )
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  z-index: 1000;
  position: absolute;
  top: ${ ({ isLoginFormDocked }: IContainer) => isLoginFormDocked ? '-100vh' : '0' };
  left: 0;
  width: 100%;
  height: 100vh;
`
interface IContainer {
  isLoginFormDocked: boolean
}

const SiteSplash = styled.div`
  width: 100%;
  height: 100vh;
  background-color: ${ ({ userColorPrimary }: ISiteSplash) => userColorPrimary };
  color: white;
  font-family: inherit;
  overflow: hidden;
`
interface ISiteSplash {
  userColorPrimary: string
}

const SpreadsheetIcon = styled.img`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.1;
`

const RegisterFormContainer = styled.div`
  z-index: 500;
  position: fixed;
  width: 25vw;
  height: 100vh;
  top: 0;
  left: ${ ({ isRegisterFormDocked }: IRegisterFormContainer) => isRegisterFormDocked ? '100vw' : '75vw' };
  transition: left 0.25s;
  background-color: ${ ({ userColorPrimary }: IRegisterFormContainer) => userColorPrimary };
`
interface IRegisterFormContainer {
  isRegisterFormDocked: boolean
  userColorPrimary: string
}

const CallToAction = styled.div`
  display: ${ ({ isLoginFormDocked }: ICallToAction) => isLoginFormDocked ? 'block' : 'none' };
  z-index: 500;
  cursor: pointer;
  position: fixed;
  top: 0;
  right: 0;
  padding-top: 0.25rem;
  padding-right: 0.5rem;
  color: white;
  font-size: 0.75rem;
  font-style: italic;
  &:hover {
    text-decoration: underline;
  }
`
interface ICallToAction {
  isLoginFormDocked: boolean
}

const AppOverlay = styled.div`
  width: 100%;
  height: 100vh;
  background-color: transparent;
  pointer-events: none;
`

const Header = styled.div`
  z-index: 10;
  position: relative;
  width: 100%;
  padding: 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  @media (max-width: 480px) {
    padding: 1rem;
    justify-content: flex-end;
  }
`

const HeaderName = styled.div`
  font-size: 1.5rem;
  font-weight: bold;
  @media (max-width: 480px) {
    display: none;
  }
`

const HeaderLinks = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-end;
`

const HeaderLink = styled.div`
  font-size: 0.9rem;
  line-height: 1.3rem;
  text-align: right;
  white-space: nowrap;
`

const Splash = styled.div`
  z-index: 10;
  position: relative;
  margin-top: 2rem;
  height: calc(100% - 5.5rem);
	display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const Name = styled.div`
  font-size: 3rem;
  font-weight: bold;
  margin-top: 10%;
  margin-bottom: 0.625rem;
  display: flex;
  align-items: flex-start;
  @media (max-width: 480px) {
    margin-top: -10%;
  }
`

const Motto = styled.div`
  font-size: 1.125rem;
  width: 80%;
  text-align: center;
`

const LoginRegisterContainer = styled.div`
	display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1rem;
  @media (max-width: 480px) {
    width: 80%;
  }
`

const Divider = styled.div`
  margin: 2rem 0;
  width: 10rem;
  height: 1px;
  background-color: white;
`



export default Site
