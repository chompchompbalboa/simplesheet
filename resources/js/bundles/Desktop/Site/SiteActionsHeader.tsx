//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

import { CLOSE } from '@/assets/icons'

import Icon from '@/components/Icon'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SiteActionsHeader = ({
  dockSiteActions
}: ISiteActionsHeader) => {
  return (
    <Container>
      <Name>todo<Gray>sheet</Gray></Name>
      <CloseSiteActions
        onClick={() => dockSiteActions()}>
        <Icon 
          size="1.5rem"
          icon={CLOSE}/>
      </CloseSiteActions>
    </Container>
  )
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface ISiteActionsHeader {
  dockSiteActions(): void
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  padding: 0 1.5rem;
  margin-bottom: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
`

const Gray = styled.span`
  color: rgb(175, 175, 175);
`

const Name = styled.div`
  font-size: 1.5rem;
  font-weight: bold;
  letter-spacing: auto;
`

const CloseSiteActions = styled.div`
  cursor: pointer;
  position: relative;
  top: -0.75rem;
  left: 2rem;
`

export default SiteActionsHeader
