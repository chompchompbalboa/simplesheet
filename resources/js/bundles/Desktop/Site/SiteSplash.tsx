//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'

import { IAppState } from '@/state'

import SiteLoginForm from '@desktop/Site/SiteLoginForm'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SiteSplash = ({
  isSiteSplashDocked
}: ISiteSplash) => {
  
  // Redux
  const userColorPrimary = useSelector((state: IAppState) => state.user.color.primary)

  return (
    <Container
      isSiteSplashDocked={isSiteSplashDocked}
      userColorPrimary={userColorPrimary}>
      <Header>
        <HeaderName>
          tasksheet
        </HeaderName>
        <HeaderLinks>
          <HeaderLink>30-day free trial<br/>$5 per month or $100 for lifetime access</HeaderLink>
        </HeaderLinks>
      </Header>
      <Splash>
        <Name>tasksheet</Name>
        <Motto>Flexible and powerful, Tasksheet is the spreadsheet specifically built for task management</Motto>
        <Divider />
        <LoginContainer>
          <SiteLoginForm
            inputsMarginLeft="0.375rem"
            statusTextAlign="center"/>
        </LoginContainer>
      </Splash>
      <SpreadsheetIcon
        src={environment.assetUrl + 'images/spreadsheet.png'}/>
    </Container>
  )
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface ISiteSplash {
  isSiteSplashDocked: boolean
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  width: 100%;
  height: 100vh;
  background-color: ${ ({ userColorPrimary }: IContainer) => userColorPrimary };
  color: white;
  font-family: inherit;
  overflow: hidden;
  border-bottom: 1px solid rgb(150, 150, 150);
  box-shadow: ${ ({ isSiteSplashDocked }: IContainer) => isSiteSplashDocked ? 'none' : '0px 3px 20px 0px rgba(0,0,0,0.75)'};
`
interface IContainer {
  isSiteSplashDocked: boolean
  userColorPrimary: string
}

const SpreadsheetIcon = styled.img`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.1;
`

const Header = styled.div`
  z-index: 10;
  position: relative;
  width: 100%;
  padding: 2rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  @media (max-width: 480px) {
    padding: 1rem;
    justify-content: flex-end;
  }
`

const HeaderName = styled.div`
  font-size: 1.5rem;
  font-weight: bold;
  @media (max-width: 480px) {
    display: none;
  }
`

const HeaderLinks = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-end;
`

const HeaderLink = styled.div`
  font-size: 0.9rem;
  line-height: 1.3rem;
  text-align: right;
  white-space: nowrap;
`

const Splash = styled.div`
  z-index: 10;
  position: relative;
  margin-top: 2rem;
  height: calc(100% - 5.5rem);
	display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const Name = styled.div`
  font-size: 3rem;
  font-weight: bold;
  margin-top: 10%;
  margin-bottom: 0.625rem;
  display: flex;
  align-items: flex-start;
  @media (max-width: 480px) {
    margin-top: -10%;
  }
`

const Motto = styled.div`
  font-size: 1.125rem;
  width: 80%;
  text-align: center;
`

const LoginContainer = styled.div`
  width: 40rem;
	display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1rem;
  @media (max-width: 480px) {
    width: 80%;
  }
`

const Divider = styled.div`
  margin: 2rem 0;
  width: 10rem;
  height: 1px;
  background-color: white;
`

export default SiteSplash
