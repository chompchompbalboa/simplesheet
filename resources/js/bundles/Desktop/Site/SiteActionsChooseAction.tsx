//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

import { IActiveSiteAction } from '@desktop/Site/SiteActions'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SiteActionsChooseAction = ({
  activeSiteAction,
  setActiveSiteAction
}: ISiteActionsChooseAction) => {
  return (
    <Container>
      <ActionLink
        isActiveSiteAction={activeSiteAction === 'REGISTER'}
        onClick={() => setActiveSiteAction('REGISTER')}>
        Register
      </ActionLink>
      <ActionLinkDivider />
      <ActionLink
        isActiveSiteAction={activeSiteAction === 'LOGIN'}
        onClick={() => setActiveSiteAction('LOGIN')}>
        Login
      </ActionLink>
    </Container>
  )
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface ISiteActionsChooseAction {
  activeSiteAction: IActiveSiteAction,
  setActiveSiteAction(nextActiveSiteAction: IActiveSiteAction): void
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  padding: 1rem;
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: rgb(250, 250, 250);
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
`

const ActionLink = styled.div`
  cursor: pointer;
  width: 25%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 0.9rem;
  font-weight: bold;
  text-decoration: ${ ({ isActiveSiteAction }: IActionLink ) => isActiveSiteAction ? 'underline' : 'none' };
  &:hover {
    text-decoration: underline;
  }
`
interface IActionLink {
  isActiveSiteAction: boolean
}

const ActionLinkDivider = styled.div`
  height: 1rem;
  width: 1px;
  background-color: black;
`

export default SiteActionsChooseAction
