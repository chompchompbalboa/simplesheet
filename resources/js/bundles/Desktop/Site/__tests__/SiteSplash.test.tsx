//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import axiosMock from 'axios'

import { renderWithRedux } from '@/testing/library'
import { 
  mockEnvironment
} from '@/testing/mocks'
import { appStateFactory, IAppStateFactoryInput } from '@/testing/mocks/appState'

import { SiteSplash } from '@desktop/Site/SiteSplash'

//-----------------------------------------------------------------------------
// Setup
//-----------------------------------------------------------------------------
const {
  allFiles,
  allFileIds,
  allSheetsFromDatabase
} = appStateFactory({} as IAppStateFactoryInput)

const fileId = allFileIds[0]
const sheetId = allFiles[fileId].typeId
const sheetFromDatabase = allSheetsFromDatabase[sheetId]

// @ts-ignore
axiosMock.get.mockResolvedValue({ data: sheetFromDatabase })

console.warn = jest.fn()
console.error = jest.fn()

//-----------------------------------------------------------------------------
// Tests
//-----------------------------------------------------------------------------
describe('SiteSplash', () => {

  beforeAll(() => {
    // @ts-ignore
    global.environment = mockEnvironment
  })
  
  it("displays a link to the registration form", async () => {
    const { queryByText } = renderWithRedux(<SiteSplash />)
    expect(queryByText('Register')).toBeTruthy()
  })
  
  it("displays a link to the login form", async () => {
    const { queryByText } = renderWithRedux(<SiteSplash />)
    expect(queryByText('Login')).toBeTruthy()
  })
  
  it("displays the registration form by default", async () => {
    const { queryByText } = renderWithRedux(<SiteSplash />)
    expect(queryByText('Sign Up')).toBeTruthy()
  })
  
  it("displays the login form after clicking the login link", async () => {
    const { getByText, queryByText } = renderWithRedux(<SiteSplash />)
    const loginLink = getByText('Login')
    expect(queryByText('Sign Up')).toBeTruthy()
    expect(queryByText('Log In')).toBeNull()
    loginLink.click()
    expect(queryByText('Sign Up')).toBeNull()
    expect(queryByText('Log In')).toBeTruthy()
  })
  
  it("displays the registration form after clicking the register link", async () => {
    const { getByText, queryByText } = renderWithRedux(<SiteSplash />)
    const loginLink = getByText('Login')
    const registerLink = getByText('Register')
    expect(queryByText('Sign Up')).toBeTruthy()
    expect(queryByText('Log In')).toBeNull()
    loginLink.click()
    expect(queryByText('Sign Up')).toBeNull()
    expect(queryByText('Log In')).toBeTruthy()
    registerLink.click()
    expect(queryByText('Sign Up')).toBeTruthy()
    expect(queryByText('Log In')).toBeNull()
  })

})
