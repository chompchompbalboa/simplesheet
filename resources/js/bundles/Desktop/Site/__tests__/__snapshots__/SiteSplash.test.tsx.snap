// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteSplash renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c15 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(250,250,250);
  border-top-left-radius: 6px;
  border-top-right-radius: 6px;
}

.c16 {
  cursor: pointer;
  width: 25%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.9rem;
  font-weight: bold;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c16:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c18 {
  cursor: pointer;
  width: 25%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.9rem;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c18:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c17 {
  height: 1rem;
  width: 1px;
  background-color: black;
}

.c27 {
  width: 100%;
  margin-top: 0.5rem;
  margin-left: 0;
  cursor: pointer;
  padding: 0.5rem 1.25rem;
  border: 1px solid rgb(150,150,150);
  border-radius: 5px;
  font-size: 0.9rem;
  background-color: rgba(220,220,220,1);
  color: black;
  outline: none;
  -webkit-transition: background-color 0.1s;
  transition: background-color 0.1s;
}

.c27:hover {
  background-color: white;
  color: black;
}

.c21 {
  margin-left: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c22 {
  width: 100%;
}

.c23 {
  width: 100%;
  margin: 0.375rem;
  padding: 0.5rem 0.25rem;
  border: none;
  border: 1px solid rgb(150,150,150);
  border-radius: 4px;
  outline: none;
  font-size: 0.9rem;
}

.c28 {
  margin-top: 0.5rem;
  text-align: left;
  color: rgb(175,0,0);
}

.c24 {
  padding: 0.25rem 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c26 {
  margin-left: 0.5rem;
  width: 100%;
}

.c25 {
  margin: 0.375rem;
  margin-left: 0;
  padding: 0.5rem 0.25rem;
  border-radius: 4px;
  font-size: 0.9rem;
}

.c20 {
  width: 100%;
}

.c0 {
  width: 100%;
  height: 100vh;
  background-color: userColorPrimary;
  color: white;
  font-family: inherit;
  overflow: hidden;
  border-bottom: 1px solid rgb(150,150,150);
}

.c3 {
  color: rgb(190,190,190);
}

.c29 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.1;
}

.c1 {
  z-index: 10;
  position: relative;
  width: 100%;
  padding: 3rem;
  padding-bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  font-size: 2rem;
  font-weight: bold;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  font-size: 0.9rem;
  line-height: 1.3rem;
  text-align: right;
  white-space: nowrap;
}

.c6 {
  z-index: 10;
  width: 100%;
  height: calc(100% - 5.5rem);
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  width: 100%;
  font-size: 2.5rem;
  font-weight: bold;
  margin-bottom: 1.5rem;
}

.c9 {
  font-size: 1.125rem;
  width: 90%;
}

.c7 {
  height: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 45%;
  margin-right: 5%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  height: 80%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 35%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  width: 90%;
  padding: 1rem;
  color: black;
}

.c19 {
  padding: 1.5rem;
  background-color: rgb(230,230,230);
  border-bottom-left-radius: 6px;
  border-bottom-right-radius: 6px;
}

.c11 {
  position: absolute;
  bottom: 5rem;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c12 {
  margin-left: 0.5rem;
  font-size: 0.8rem;
}

.c10 {
  font-weight: bold;
  white-space: nowrap;
}

@media (max-width:480px) {
  .c27 {
    width: 100%;
    margin: 0.375rem;
  }
}

@media (max-width:480px) {
  .c23 {
    width: 100%;
    margin: 0.375rem 0;
  }
}

@media (max-width:480px) {
  .c25 {
    width: 100%;
    margin: 0.375rem 0;
  }
}

@media (max-width:480px) {
  .c1 {
    padding: 1rem;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (max-width:480px) {
  .c2 {
    display: none;
  }
}

@media (max-width:480px) {
  .c8 {
    margin-top: -10%;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            task
            <span
              class="c3"
            >
              sheet
            </span>
          </div>
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              30-day free trial
              <br />
              $5 per month or $100 for lifetime access
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <div
            class="sc-fzXfMB c7"
          >
            <div
              class="c8"
            >
              The spreadsheet that's perfect
              <br />
              for your workplace to-do lists
            </div>
            <div
              class="c9"
            >
              <span
                class="c10"
              >
                Tasksheet
              </span>
               is a spreadsheet fine-tuned for task management. Built around its ability to quickly and easily organize your to-dos, it is a uniquely flexible and powerful solution for managing life at work
            </div>
            <div
              class="c11"
            >
              <div
                class=""
              >
                <svg
                  height="NaN"
                  viewBox="0 0 1024 1024"
                  width="NaN"
                >
                  <path
                    d="M854 512l-342 342-342-342 62-60 238 238v-520h84v520l240-238z"
                    style="fill: currentColor;"
                  />
                </svg>
              </div>
              <div
                class="c12"
              >
                Scroll down to try it out
              </div>
            </div>
          </div>
          <div
            class="sc-fzXfMB c13"
          >
            <div
              class="c14"
            >
              <div
                class="c15"
              >
                <div
                  class="c16"
                >
                  Register
                </div>
                <div
                  class="c17"
                />
                <div
                  class="c18"
                >
                  Login
                </div>
              </div>
              <div
                class="c19"
              >
                <form
                  class="c20"
                >
                  <div
                    class="c21"
                  >
                    <label
                      class="c22"
                    >
                      Name
                      :
                    </label>
                    <input
                      class="c23"
                      placeholder="Name"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="c21"
                  >
                    <label
                      class="c22"
                    >
                      Email
                      :
                    </label>
                    <input
                      class="c23"
                      placeholder="Email"
                      type="email"
                      value=""
                    />
                  </div>
                  <div
                    class="c21"
                  >
                    <label
                      class="c22"
                    >
                      Password
                      :
                    </label>
                    <input
                      class="c23"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    class="c24"
                  >
                    <input
                      class="c25"
                      id="SiteFormCheckbox"
                      type="checkbox"
                    />
                    <label
                      class="c26"
                      for="SiteFormCheckbox"
                    >
                      I agree to start a 30-day free trial of Tasksheet (no credit card required)
                    </label>
                  </div>
                  <button
                    class="c27"
                  >
                    Sign Up
                  </button>
                  <div
                    class="c28"
                    data-testid="SiteSplashRegisterFormStatus"
                  />
                </form>
              </div>
            </div>
          </div>
        </div>
        <img
          class="c29"
          src="testAssetUrlimages/spreadsheet.png"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzXfLU jvpYiT"
    >
      <div
        class="sc-fzXfLX hedHvX"
      >
        <div
          class="sc-fzXfMv grQEfd"
        >
          task
          <span
            class="sc-fzXfLV eVfinF"
          >
            sheet
          </span>
        </div>
        <div
          class="sc-fzXfMw ivjzrb"
        >
          <div
            class="sc-fzXfMx eBnBNG"
          >
            30-day free trial
            <br />
            $5 per month or $100 for lifetime access
          </div>
        </div>
      </div>
      <div
        class="sc-fzXfMy fpkgyZ"
      >
        <div
          class="sc-fzXfMB sc-fzXfMC kHlPLJ"
        >
          <div
            class="sc-fzXfMz ceRLfX"
          >
            The spreadsheet that's perfect
            <br />
            for your workplace to-do lists
          </div>
          <div
            class="sc-fzXfMA gqURvr"
          >
            <span
              class="sc-fzXfNg ivscKY"
            >
              Tasksheet
            </span>
             is a spreadsheet fine-tuned for task management. Built around its ability to quickly and easily organize your to-dos, it is a uniquely flexible and powerful solution for managing life at work
          </div>
          <div
            class="sc-fzXfNd cKdYfo"
          >
            <div
              class="sc-fzXfNf gOUNtp"
            >
              <svg
                height="NaN"
                viewBox="0 0 1024 1024"
                width="NaN"
              >
                <path
                  d="M854 512l-342 342-342-342 62-60 238 238v-520h84v520l240-238z"
                  style="fill: currentColor;"
                />
              </svg>
            </div>
            <div
              class="sc-fzXfNe cLRbAW"
            >
              Scroll down to try it out
            </div>
          </div>
        </div>
        <div
          class="sc-fzXfMB sc-fzXfMD cjxjTH"
        >
          <div
            class="sc-fzXfME SEwbI"
          >
            <div
              class="sc-AykKC iprWqD"
            >
              <div
                class="sc-AykKD bWmehU"
              >
                Register
              </div>
              <div
                class="sc-AykKE jSIRJj"
              />
              <div
                class="sc-AykKD ircMQC"
              >
                Login
              </div>
            </div>
            <div
              class="sc-fzXfNc gCrwon"
            >
              <form
                class="sc-fzXfLT gveSwo"
              >
                <div
                  class="sc-AykKG kgqMhc"
                >
                  <label
                    class="sc-AykKH bXEuhu"
                  >
                    Name
                    :
                  </label>
                  <input
                    class="sc-AykKI eNDOHp"
                    placeholder="Name"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="sc-AykKG kgqMhc"
                >
                  <label
                    class="sc-AykKH bXEuhu"
                  >
                    Email
                    :
                  </label>
                  <input
                    class="sc-AykKI eNDOHp"
                    placeholder="Email"
                    type="email"
                    value=""
                  />
                </div>
                <div
                  class="sc-AykKG kgqMhc"
                >
                  <label
                    class="sc-AykKH bXEuhu"
                  >
                    Password
                    :
                  </label>
                  <input
                    class="sc-AykKI eNDOHp"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                </div>
                <div
                  class="sc-fzXfLQ hFeOtt"
                >
                  <input
                    class="sc-fzXfLS dmVpOr"
                    id="SiteFormCheckbox"
                    type="checkbox"
                  />
                  <label
                    class="sc-fzXfLR gXjWxK"
                    for="SiteFormCheckbox"
                  >
                    I agree to start a 30-day free trial of Tasksheet (no credit card required)
                  </label>
                </div>
                <button
                  class="sc-AykKF gFNNfV"
                >
                  Sign Up
                </button>
                <div
                  class="sc-AykKJ jJgrwI"
                  data-testid="SiteSplashRegisterFormStatus"
                />
              </form>
            </div>
          </div>
        </div>
      </div>
      <img
        class="sc-fzXfLW brNufP"
        src="testAssetUrlimages/spreadsheet.png"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
